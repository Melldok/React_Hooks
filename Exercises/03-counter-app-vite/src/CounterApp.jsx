import React from 'react'
import PropTypes from 'prop-types';
import { useState } from 'react';

export const CounterApp = ({value}) => {

    //useState Hook
    const [counter, setCounter] = useState( value )

    const handleIncrement = () => {
            
        setCounter(counter + 1)
    }

    const handleDecrement = () => {
        
        setCounter(counter - 1)
    
    }

    const handleReset = () => {
        
        setCounter(value)
    
    }
                                            


    
  return (
    <>
    <h1>CounterApp</h1>
    <h2 data-testid='test-counter'> { counter } </h2>
    <button onClick={handleIncrement} >
        Increment
    </button>
    <button onClick={handleDecrement}>
        Decrement
    </button>

    <button onClick={handleReset}>
        Reset
    </button>
    </>
  )
}


CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}
