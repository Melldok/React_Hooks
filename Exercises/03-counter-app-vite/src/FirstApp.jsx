import React from 'react'
import PropTypes from 'prop-types';

export const FirstApp = ( {title, subtitle, name} ) => {

    

    return (
        <> 
            <div data-testid='test-title'>{title}</div>
            <p>{subtitle}</p>
            <p>{subtitle}</p>
            <p> {name} </p>
        </>
  )
}


// Show the usage of your component by setting property types
FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
}

// If there are no speficication on the props, use these defaults
FirstApp.defaultProps = {
    
    subtitle: 'There is no subtitle',
    name: 'There is no name'
}