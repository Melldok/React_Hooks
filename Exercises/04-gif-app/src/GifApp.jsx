
import React from 'react';
import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifApp = () => {

    const [categories, setCategories] = useState([])

    const onAddCategory = (newCategory) => {
        
        // Case sensitive array validation

        const isContained = categories.some(category => {
            return category.toLowerCase() === newCategory.toLowerCase();
          });
          

        if(isContained) return;
       
        setCategories([...categories, newCategory])

    }

    


  return (
    <>
        {/* Title */}
        <h1>GifApp</h1>

        {/* Input */}
        <AddCategory 
        //setCategories={setCategories}
            onNewCategory={onAddCategory}
        />
        {/* Gif Listing */}
        <ol>
            {
            categories.map((category) => (
                <GifGrid key={category} category={category}/>
            ))
            }
        </ol>
            {/* Gif item */}
    </>
  )
}
