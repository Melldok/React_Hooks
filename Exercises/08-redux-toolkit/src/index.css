#root{
  text-align: start;
}

body{
  margin: 0;
  padding: 50px;
  text-align: left;
}

