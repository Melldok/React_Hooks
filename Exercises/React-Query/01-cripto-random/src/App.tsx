import './App.css'
import { useQuery } from '@tanstack/react-query'
import { useRandom } from './hooks/useRandom'



export const App = () => {
  
    // Mejor tener la lógica de useQuery separada en custom hooks, el código será más mantenible, ya que si hay algún cambio en la lógica de la llamada a la API, solo tendremos que cambiar el custom hook.
    const query = useRandom();
  
    return (
      <>
        <div className='App App-header'>
          
          {query.isFetching ? 
            <h2>Cargando...</h2> : 
            <h2>Número aleatorio : {query.data}</h2>
          }
  
          {
            !query.isLoading && query.isError && (<h3>{`${query.error}`}</h3>)
          }

          {/* Utilizamos isFetching porque isLoading es true cuando se está cargando por primera vez. isFetching es true cuando se está refrescando la información. */}

          <button onClick={() => query.refetch()} disabled={query.isFetching}>
            {
                query.isFetching ? '...' : 'Nuevo Número'
            }
          </button>
           
  
        </div>
      </>
    )
  }
    
    
  
        

 


