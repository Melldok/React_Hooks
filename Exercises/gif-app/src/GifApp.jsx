
import React from 'react';
import { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifApp = () => {

    const [categories, setCategories] = useState(["One Punch"])

    const onAddCategory = (newCategory) => {
        
        // Case sensitive array validation

        const isContained = categories.some(category => {
            return category.toLowerCase() === newCategory.toLowerCase();
          });

        if(isContained) return;
       
        setCategories([...categories, newCategory])

    }

    


  return (
    <>
        {/* Title */}
        <h1>GifApp</h1>

        {/* Input */}
        <AddCategory 
        //setCategories={setCategories}
            onNewCategory={onAddCategory}
        />
        {/* Gif Listing */}
        <ol>
            {
            categories.map((category) => (
                <GifGrid key={category} category={category}/>
            ))
            }
        </ol>
            {/* Gif item */}
    </>
  )
}
