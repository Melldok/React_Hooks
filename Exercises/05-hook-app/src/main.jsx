import React from 'react'
import ReactDOM from 'react-dom/client'
import { HooksApp } from './HooksApp'
import { CounterApp } from './useState/CounterApp'
import { CounterWithCustomHook } from './useState/CounterWithCustomHook'



import './index.css'
import { SimpleForm } from './useEffect/SimpleForm'
import { FormWithCustomHook } from './useEffect/FormWithCustomHook'
import { MultipleCustomHooks } from './examples/MultipleCustomHooks'
import { FocusScreen } from './useRef/FocusScreen'
import { Layout } from './useLayoutEffect/Layout'
import { Memorize } from './memos/Memorize'
import { MemoHook } from './memos/MemoHook'
import { CallbackHook } from './memos/CallbackHook'
import { Father } from './07-tarea-memo/Father'
// import './useReducer/intro-reducer'
import { TodoApp } from './useReducer/TodoApp'
import { MainApp } from './useContext/MainApp'
import { BrowserRouter } from 'react-router-dom'




ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
  
  <BrowserRouter>
    <MainApp />
  </BrowserRouter>
  
  //</React.StrictMode>
)