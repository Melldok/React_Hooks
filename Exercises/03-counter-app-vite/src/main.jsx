import React from 'react';
import ReactDOM from 'react-dom/client';
import { App } from './App'
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import './style.css'
CounterApp



ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <CounterApp value={0}></CounterApp>
    </React.StrictMode>
)