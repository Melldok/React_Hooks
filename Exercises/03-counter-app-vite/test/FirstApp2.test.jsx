import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";




describe('Testing on FirstApp component', () => { 

    const title = 'Hey, I am Robb'
    const subtitle = 'I am a subtitle'
   
    test('should match the snapshot', () => { 
       
        

        const { container } = render ( <FirstApp title={ title } />)
        expect( container ).toMatchSnapshot();

     });
    
    test('should show message "Hi, I am Robb"', () => { 
        
        render ( <FirstApp title={ title } />)

        //Screen represents the whole component we are rendering.
        expect( screen.getByText(title) ).toBeTruthy();
    });

    test('should show the title in the H1', () => { 

        render ( <FirstApp title={ title } />)
        expect( screen.getByRole('heading', {level : 1}).innerHTML).toContain(title);

     });

     test('should show subtitle sent by props', () => { 

        render ( 
        
        <FirstApp 
            
            title={ title }
            subtitle={ subtitle }
                
        />);
       
        expect( screen.getAllByText(subtitle).length).toBe(2);

      })


 })