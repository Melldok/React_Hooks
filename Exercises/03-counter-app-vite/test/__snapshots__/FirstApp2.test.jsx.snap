// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on FirstApp component should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hey, I am Robb
  </h1>
  <p>
    There is no subtitle
  </p>
  <p>
    There is no subtitle
  </p>
  <p>
     
    There is no name
     
  </p>
</div>
`;
